app/models/episode.rb

```diff
@@ -133,28 +129,75 @@ def author=(a)
     self.author_email = author["email"]
   end
 
-  def enclosure
-    enclosures.complete.first
+  def ready_image
+    images.complete_or_replaced.first
   end
 
   def image
-    images.complete.first
-  end
-
-  # API updates for image=
-  def image_file
     images.first
   end
 
-  def image_file=(file)
+  def image=(file)
     img = EpisodeImage.build(file)
-    if img && img.original_url != image_file.try(:original_url)
-      images << img
-    elsif !img
-      images.destroy_all
+
+    if !img
+      images.each(&:mark_for_destruction)
+    elsif img&.replace?(image)
+      images.build(img.attributes.compact)
+    else
+      img.update_image(image)
+    end
+  end
+
+  def ready_enclosure
+    enclosures.complete_or_replaced.first
+  end
+
+  def enclosure
+    enclosures.first
+  end
+
+  def enclosure=(file)
+    enc = Enclosure.build(file)
+
+    if !enc
+      enclosures.each(&:mark_for_destruction)
+    elsif enc&.replace?(enclosure)
+      enclosures.build(enc.attributes.compact)
+    else
+      enc.update_resource(enclosure)
     end
   end
 
+  def ready_contents
+    contents.complete_or_replaced.group_by(&:position).values.map(&:first)
+  end
+
+  def contents=(files)
+    existing = contents.group_by(&:position)
+
+    files&.each_with_index do |file, idx|
+      position = idx + 1
+      con = Content.build(file, position)
+
+      if !con
+        existing[position]&.each(&:mark_for_destruction)
+      elsif con.replace?(existing[position]&.first)
+        contents.build(con.attributes.compact)
```

Comment:
Trying hard not to take actions that will persist changes right now.  So got rid of any `<<` on the collection, or `destroy_all` calls.

Reply:
No - new contents start out as `created`.  (And we don't allow manually setting the status, via API or worker).

Reply:
Hmm... you're asking what happens if we replace this position with a `Content` that already exists (somehow was saved), and has been processed by Porter?

I'm not sure how that could happen logically (I _think_ Content have to be assigned to an Episode to be created?)... but this code would copy its `original_url` again to a brand new Content (`status: 'created'`), and re-process it.